### PR TITLE
TEVA-1414 Improve database backup and restore scripts

### PR DESCRIPTION
**Pull request by nickromneydfe**
https://dfedigital.atlassian.net/browse/TEVA-1414

## Changes in this PR:

- Split out actions into more granular scripts
- Added workflow to restore an existing backup from S3 to staging or dev
- Use custom DFE actions to install CF CLI and Postgres client

## Next steps:

- [x] Merge https://github.com/DFE-Digital/github-actions/pull/14 to create custom GH Actions
- [x] Terraform apply S3 IAM permissions to list buckets and download files

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2729.london.cloudapps.digital>

**Review by csutter — APPROVED** [review]
👍

**Review by cesidio — APPROVED** [review]
LGTM

File: db/scripts/sanitise.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+       SET email=concat('anonymised-jobseeker-',id,'@example.org'),
+           encrypted_password='ABCDEFGH12345',
+           reset_password_token='12345ABCDEFGH',
+           current_sign_in_ip='8.8.8.8',
+           last_sign_in_ip='8.8.4.4',
+           confirmation_token='CDEFGHIJ34567',
+           unconfirmed_email=concat('anonymised-jobseeker-',id,'@example.org'),
+           unlock_token='34567CDEFGHIJ'
+       WHERE email NOT LIKE '%@digital.education.gov.uk'
+             AND email NOT LIKE '%@education.gov.uk';
```
Comment (david-mears-dfe) [inline db/scripts/sanitise.sql:22]: Could be one line? `WHERE email NOT LIKE '%education.gov.uk';`

Comment (nickromneydfe) [reply db/scripts/sanitise.sql]: Yes, that's a helpful improvement. Thanks.

**Pull request merged**